insalan/payment/views.py

```diff
@@ -0,0 +1,203 @@
+"""Views for the Payment module"""
+
+import json
+import logging
+
+import requests
+
+from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned
+from django.http import JsonResponse, HttpResponseRedirect
+from django.shortcuts import render
+from django.utils.translation import gettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework import generics, permissions, status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authentication import SessionAuthentication
+
+import insalan.settings as app_settings
+import insalan.payment.serializers as serializers
+
+from .models import Transaction, TransactionStatus, Product
+from .tokens import Token
+
+logger = logging.getLogger(__name__)
+
+
+class ProductList(generics.ListAPIView):
+    paginator = None
+    serializer_class = serializers.ProductSerializer
+    queryset = Product.objects.all()
+    permission_classes = [permissions.IsAdminUser]
+
+
+class ProductDetails(generics.RetrieveUpdateDestroyAPIView):
+    paginator = None
+    serializer_class = serializers.ProductSerializer
+    queryset = Product.objects.all()
+    permission_classes = [permissions.IsAdminUser]
+
+
+class TransactionList(generics.ListAPIView):
+    paginator = None
+    serializer_class = serializers.TransactionSerializer
+    queryset = Transaction.objects.all().order_by("last_modification_date")
+    permission_classes = [permissions.IsAdminUser]
+
+
+class TransactionPerId(generics.RetrieveAPIView):
+    paginator = None
+    serializer_class = serializers.TransactionSerializer
+    queryset = Transaction.objects.all()
+    permission_classes = [permissions.IsAdminUser]
+
+
+class CreateProduct(generics.CreateAPIView):
+    serializer_class = serializers.ProductSerializer
+    queryset = Product.objects.all()
+    permission_classes = [permissions.IsAdminUser]
+
+
+class Notifications(APIView):
+    """
+    Notifications view
+    """
+    def post(self, request):
```

Comment:
Should the logic from `ReturnView` come here then?

Reply:
Yep, that's the plan. I pushed a very small version now so I can print-debug on my VPS (which I will do tomorrow because I just spent an hour fixing two little tests and I'm tired)

Reply:
Done in https://github.com/InsaLan/backend-insalan.fr/pull/67/commits/a62eab248dd6f5a23fdabf4e60efa47ce3e62ac5